### ISSUE TITLE
runtime: performance regression due to bad instruction used in morestack_noctxt for ppc64 in CL 425396

### ISSUE THREAD

**Issue by laboger** [golang/go#57741]
### What version of Go are you using (`go version`)?

<pre>
$ go version
latest
</pre>

### Does this issue reproduce with the latest release?

yes

### What operating system and processor architecture are you using (`go env`)?

<details><summary><code>go env</code> Output</summary><br><pre>
$ go env
ppc64le power10
</pre></details>

### What did you expect to see?
Consistent performance of runtime benchmarks.

### What did you see instead?
Some benchmarks in the runtime package show a 3X or more degradation on power10 after CL 425396 was merged. Removing this line in the latest also removes the degradation. For example, Hash65536 is one example.

```
name          old time/op    new time/op   delta
Hash65536-64    4.08µs ± 2%  18.89µs ±25%  +363.16%  (p=0.029 n=4+4)

name          old speed      new speed     delta
Hash65536-64  16.1GB/s ± 2%  3.5GB/s ±22%   -77.96%  (p=0.029 n=4+4)
```

The MOVD R1, R1 results in 
```or r1,r1,r1``` 
which is a special instruction on ppc64 that changes thread priority. This is described in section 3.2 of ISA 3.1. 

Changing it to OR $0, R1 has the same effect wrt SPWRITE but does not affect thread priority.

@cherrymui @pmur @archanaravindar

**Comment by cherrymui** [comment]
Interesting. Good finding!
Mind sending a CL? Or I could send one, but I probably won't be able to validate the performance impact myself. Thanks.

**Comment by laboger** [comment]
We can do it. We have been testing it out.

**Comment by laboger** [comment]
@cherrymui we want this in Go 1.20, so is there anything special needed on the CL now that the tree is open for 1.21?

**Comment by cherrymui** [comment]
The tree hasn't opened yet. So the CL can be submitted as usual.

**Comment by archanaravindar** [comment]
@gopherbot please backport this to previous releases Go1.19 and Go1.18
Since the fix #54332 that causes this regression on ppc64 has been likewise back ported to these two releases
